tracing/trace_state.hh

```diff
@@ -276,13 +276,14 @@ private:
      * Starts the tracing session time measurments.
      * This overload is meant for primary sessions.
      *
-     * @param request description of a request being traces
+     * @param name description of a tracing context.
+     *             Caller must guarantee static storage duration.
      * @param client address of a client the traced request came from
      */
-    void begin(sstring request, gms::inet_address client) {
+    void begin(const char* name, gms::inet_address client) {
         begin();
         _records->session_rec.client = client;
-        _records->session_rec.request = std::move(request);
+        _records->session_rec.name = name;
```

Comment:
This is a *very* bogus and dangerous API that completely breaks a RAI principle. 
And it is going to cause use-after-free if someone is going to use it as below for instance.
```
tracing::begin(format("Some fancy value: {} and another fancy value: {}", val1, val2), addr);
```

Let's not create such bad APIs, please.

Reply:
@vladzcloudius are you sure your example works? I don't think there's an automatic conversion of std::string or sstring (the result of format()) into a "const char *" - you would need to call the `c_str()` method. And no self-respecting C++ programmer will take the `c_str()` of a temporary variable (the result of format()) and expect it to work.
Anyway, beyond other ideas I suggested earlier to enforce string-literals-only, this stackoverflow question https://stackoverflow.com/questions/13724242/c11-function-that-only-accepts-string-literals has a way to check that a function gets only string literals with some ugly template trick. You could make the `begin(char*)` function private, and have a public template function like this that only takes an array as in that example.

Reply:
@nyh stop looking at the string you currently pass to begin() as a string the user is supposed to be controlling - it's a tag and the user needs to have no control over it as I explained in my other post.

Once you realize that you'll see that enum is the proper way to move forward.